sql/select.go

```diff
@@ -318,126 +320,91 @@ func (v *indexInfo) analyzeOrdering(scan *scanNode, ordering []int) {
 	}
 }
 
-// makeStartInfo populates the indexInfo.start slice for the index using the
-// supplied expression. The start info contains the start values for a scan of
-// the index. As such, the construction looks for =, >= and > comparisons of
-// the index columns to literal values.
-func (v *indexInfo) makeStartInfo(exprs []parser.Exprs) {
+// makeConstraints populates the indexInfo.constraints field based on the
+// analyzed expressions. The constraints are a start and end expressions for a
+// prefix of the columns that make up the index. For example, consider an index
+// on the columns (a, b, c). For the expressions "a > 1 AND b > 2" we would
+// have the constraints:
+//
+//   {a: {start: > 1}
+//
+// Why is there no constraint on "b"? Because the start constraint was > and
+// such a constraint does not allow us to consider further columns in the
+// index. What about the expression "a >= 1 AND b > 2":
+//
+//   {a: {start: >= 1}, b: {start: > 2}}
```

Comment:
Ah, I see those tests now. I was (and still am) surprised that the functionality added in this PR didn't influence more of them.